src/Umbraco.Examine/ContentIndexPopulator.cs

```diff
@@ -70,31 +77,83 @@ protected override void PopulateIndexes(IReadOnlyList<IIndex> indexes)
             {
                 contentParentId = _parentId.Value;
             }
+
+            if (_publishedValuesOnly)
+            {
+                IndexPublishedContent(contentParentId, pageIndex, pageSize, indexes);
+            }
+            else
+            {
+                IndexAllContent(contentParentId, pageIndex, pageSize, indexes);
+            }
+        }
+
+        protected void IndexAllContent(int contentParentId, int pageIndex, int pageSize, IReadOnlyList<IIndex> indexes)
+        {
             IContent[] content;
 
             do
             {
-                if (!_publishedValuesOnly)
-                {
-                    content = _contentService.GetPagedDescendants(contentParentId, pageIndex, pageSize, out _).ToArray();
-                }
-                else
+                content = _contentService.GetPagedDescendants(contentParentId, pageIndex, pageSize, out _).ToArray();
+
+                if (content.Length > 0)
                 {
-                    //add the published filter
-                    //note: We will filter for published variants in the validator
-                    content = _contentService.GetPagedDescendants(contentParentId, pageIndex, pageSize, out _,
-                        _publishedQuery, Ordering.By("Path", Direction.Ascending)).ToArray();
+                    // ReSharper disable once PossibleMultipleEnumeration
+                    foreach (var index in indexes)
+                        index.IndexItems(_contentValueSetBuilder.GetValueSets(content));
```

Comment:
This can be improved. There's no reason to call the same method to build value sets for each index when the result is the same for all of them: `_contentValueSetBuilder.GetValueSets(content)`. I realize this was like this before but it's a very easy win to not have to allocate a whole bunch of objects for each index. So before we enumerate we can store `var valueSets = _contentValueSetBuilder.GetValueSets(content).ToList()` and then just `index.IndexItems(valueSets);`

Reply:
Fixed it